### PR TITLE
feat: [explore] don't save filters inherited from a dashboard

### PR DESCRIPTION
**Pull request by mistercrunch**
Screen Shot 2020-03-20 at 11 30 17 PM

When navigating to explore from a dashboard context, the current
dashboard filter(s) are passed along to explore so that the context is
kept. So say you're filtering on "country=Romania", in your dashboard
and pivot to explore, that filter is still there and keep on exploring.

Now a common issue is that you'll want to make some tweak to your chart
that are unrelated to the filter, say toggling the legend off for
instance, and then save it. Now you back to your dashboard and even
though you started with an "all countries" dashboard, with a global
filter on country, now that one chart is stuck on "Romania". Typically
you notice this when filtering on something else, say "Italy" and then
that one chart now has two mutually exclusive filters, and show "No data".

Now, the fix is to flag the filter as "extra" (that's the not-so-good internal
name we use for these inherited filters) and make it clear that that
specific filter is special and won't be saved when saving the chart.

### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/core.py [hunk 1]
```diff
@@ -860,6 +861,7 @@ def get_filter_key(f):
                 existing_filters[get_filter_key(existing)] = existing["comparator"]
 
         for filtr in form_data["extra_filters"]:
+            filtr["isExtra"] = True
             # Pull out time filters/options and merge into form data
             if date_options.get(filtr["col"]):
                 if filtr.get("val"):
```
Comment (villebro) [inline superset/utils/core.py]: nit: `extra_filter` or `filter_`

**Review by villebro — APPROVED** [review]
Great feature. One small non blocking nit, LGTM.

File: superset/utils/core.py (hunk 1 above)
Comment (mistercrunch) [reply superset/utils/core.py]: I agree `filter_ ` is slightly better but being lazy and mergin' as is.
ship

**Comment by mistercrunch** [comment]
FYI did a bit of design review with @rusackas today and we decided it was ok. Discussed alternative like introducing a "Inherited Filters" control similar to `Filters`, but would be confusing if empty. Decided to roll with this for now.

**Pull request merged**